lecli/query/api.py

```diff
@@ -83,111 +83,183 @@ def continue_request(response, progress_bar):
         handle_response(new_response, progress_bar)
 
 
-def fetch_results(provided_url):
+def fetch_results(provided_url, params=None):
     """
     Make the get request to the url and return the response.
     """
     try:
-        response = requests.get(provided_url, headers=api_utils.generate_headers('rw'))
+        response = requests.get(provided_url, headers=api_utils.generate_headers('rw'),
+                                params=params)
         return response
     except requests.exceptions.RequestException as error:
-        click.echo(error)
+        click.echo(error, err=True)
         sys.exit(1)
 
 
-def get_recent_events(log_keys, last_x_seconds=1200, time_range=None):
+def validate_query(**kwargs):
     """
-    Get recent events belonging to provided log_keys in the last_x_seconds.
+    Validate query options
     """
-    if time_range:
-        leql = {"during": {"time_range": time_range}, "statement": ALL_EVENTS_QUERY}
-    else:
-        to_ts = int(time.time()) * 1000
-        from_ts = (int(time.time()) - last_x_seconds) * 1000
-        leql = {"during": {"from": from_ts, "to": to_ts}, "statement": ALL_EVENTS_QUERY}
-    payload = {"logs": log_keys, "leql": leql}
+    date_from = kwargs.get('date_from')
+    time_from = kwargs.get('time_from')
+    relative_time_range = kwargs.get('relative_time_range')
+    saved_query_id = kwargs.get('saved_query_id')
+    query_string = kwargs.get('query_string')
+    log_keys = kwargs.get('log_keys')
+    querynick = kwargs.get('querynick')
+    lognick = kwargs.get('lognick')
+    loggroup = kwargs.get('loggroup')
```

Comment:
Is this a logset? For naming consistency we should probably use the convention name

Reply:
There is a known issue with the naming of "loggroup", "lognick" and "querynick" in lecli. They are not related to shortcuts, logsets etc we have in the server side. And they are confusing at some sense. There is actually an open bug for this already. We could solve it in another PR or this one. However, we don't have the names to go for the moment.